tests/serializer/tests_Serialize.cpp

```diff
@@ -0,0 +1,181 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+#include <filesystem>
+#include <format>
+#include <random>
+
+using namespace rdf4cpp::rdf;
+
+TEST_SUITE("Serialize") {
+    template<typename NodeT>
+    void run_ser_test(NodeT node) {
+        auto expected = std::string{node};
```

Comment:
This tests the serializer by comparing with the serializer because std::string also uses a serializer. Expected should be hardcoded.